nansat/mappers/mapper_opendap_mywave.py

```diff
@@ -0,0 +1,76 @@
+# Name:         mapper_opendap_mywave.py
+# Purpose:      Nansat mapping for MyWaveWAV data provided by MET.NO
+# Author:       Artem Moiseev
+# Licence:      This file is part of NANSAT. You can redistribute it or modify
+#               under the terms of GNU General Public License, v.3
+#               http://www.gnu.org/licenses/gpl-3.0.html
+
+from nansat.mappers.opendap import Opendap
+from nansat.exceptions import WrongMapperError
+from nansat.nsr import NSR
+from netCDF4 import Dataset
+from datetime import datetime
+import numpy as np
+import json
+import pythesint as pti
+import os
+
+
+class Mapper(Opendap):
+
+    baseURLs = ['http://thredds.met.no/thredds/dodsC/fou-hi/mywavewam4archive']
+    timeVarName = 'time'
+    xName = 'rlon'
+    yName = 'rlat'
+    timeCalendarStart = '1970-01-01'
+
+    def __init__(self, filename, gdal_dataset, gdal_metadata, date=None,
+                 ds=None, bands=None, cachedir=None, *args, **kwargs):
+
+        self.test_mapper(filename)
+        timestamp = date if date else self.get_date(filename)
+        ds = Dataset(filename)
+        try:
+            self.srcDSProjection = NSR(ds.variables['projection_3'].proj4 +
+                                       ' +to_meter=0.0174532925199 +wktext')
+        except KeyError:
+            raise WrongMapperError
+
+        self.create_vrt(filename, gdal_dataset, gdal_metadata, timestamp, ds, bands, cachedir)
+        mm = pti.get_gcmd_instrument('Computer')
+        ee = pti.get_gcmd_platform('MODELS')
+        self.dataset.SetMetadataItem('instrument', json.dumps(mm))
+        self.dataset.SetMetadataItem('platform', json.dumps(ee))
+        self.dataset.SetMetadataItem('Data Center', 'NO/MET')
+        self.dataset.SetMetadataItem('Entry Title', str(ds.getncattr('title')))
+
+    @staticmethod
+    def get_date(filename):
+        """Extract date and time parameters from filename and return
+        it as a formatted string
+
+        Parameters
+        ----------
+
+        filename: str
+            nn
+
+        Returns
+        -------
+            str, YYYY-mm-ddThh:MMZ
+
+        Examples
+        --------
+            >>> Mapper.get_date('/path/to/MyWave_wam4_WAVE_20171029T18Z.nc')
+            '2017-10-29T18:00Z'
+        """
+        _, filename = os.path.split(filename)
+        t = datetime.strptime(filename.split('_')[-1], '%Y%m%dT%HZ.nc')
+        return datetime.strftime(t, '%Y-%m-%dT%H:%MZ')
```

Comment:
there is another way to do this, e.g., as in `envisat.py`:

```
from dateutil.parser import parse
...
self.dataset.SetMetadataItem('time_coverage_start', parse(gdalMetadata["SPH_FIRST_LINE_TIME"]).isoformat())
```

Reply:
I believe that the method used in my code is much simpler and explicitly shows what happens

Reply:
Maybe, but it is also repetition... Can you at least specify in the docstring and above line 69 that it is formatted in isoformat?